### ISSUE TITLE
Unable to drag on Firefox's responsive design mode (r is undefined)

### ISSUE THREAD

**Issue by rayhanw** [mapbox/mapbox-gl-js#8877]
**mapbox-gl-js version**:1.4.1 (also had the same issue on 1.4.0)

**browser**: Firefox Quantum (69.0.3)

### Steps to Trigger Behavior

 1. Switch to firefox responsive design mode
 2. **Choose a phone preset** (Only does not work on any phone preset, works fine if on responsive preset)
List of broken phone presets:
- Galaxy S9/S9+
- iPad
- iPhone 6/7/8
- iPhone 6/7/8 Plus
- iPhone X/XS
- Kindle Fire HDX
- Galaxy Note 9

### Link to Demonstration
**Don't forget to put the mapbox api key on the `script` tag in the HTML**
https://codepen.io/rayhanw/pen/poobNrY

### Expected Behavior
Content of map gets dragged around
### Actual Behavior
- Getting an `r is undefined` error when trying to drag the map
- Map does not get dragged
![mapbox](https://user-images.githubusercontent.com/20383179/66901202-d68a6000-f030-11e9-8c33-977edd36103d.gif)

**Comment by arindam1993** [comment]
This looks like a bug with firefox's touch event simulator, its sending and invalid `TouchEvent` instance that does not have the `touches` array.

**Comment by kkaefer** [comment]
I confirmed with a physical iOS devices that it does indeed get touches. In addition, Firefox seems to add a touch to the `TouchList` object at some point _after_ calling `DOM.mousePos`: When using `console.log`, the `TouchList` objects show up as empty, but when expanding them, the actually do contain the touch.

https://bugzilla.mozilla.org/show_bug.cgi?id=1588438 looks like it might be related.

**Comment by kkaefer** [comment]
I can also reproduce https://bugzilla.mozilla.org/show_bug.cgi?id=1549220: when adding a `console.log(e)` before acessing `e.touches[0]`, the `TouchList` is correctly populated and the error goes away.

**Comment by kkaefer** [comment]
I don't think we should add a workaround for this, given that the Responsive Design Mode is a developer tool and end users are unlikely to encounter this.
